### PR TITLE
SCF 29 Turn Searchbar Into a Component

### PR DESCRIPTION
**Pull request by jeyu21**
SCF-29. I made searchbar a component and added it to the components/layout folder. 
I changed the searchbar in Catalog, Bookmarks, and AppTracker
I made it so that the searchbar becomes width: 100% after the page becomes smaller.

https://user-images.githubusercontent.com/21046037/140906894-8aec90fd-3f38-4f73-b55d-5e7baee46773.mov

https://user-images.githubusercontent.com/21046037/140907012-ea99c0a6-c4d0-4afd-b406-e3fa40e32c4b.mov

https://user-images.githubusercontent.com/21046037/140907097-611d6d20-acbc-4ee7-ab0e-9d6793df49bf.mov

The modal searchbar looks kinda sus and I was trying to observe the CSS but I didn't spot it. Would like a second eye on this

### CODE REVIEW

File: src/components/layout/searchbar/Searchbar.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        <input
+          type="text"
+          placeholder="Search"
+          value={value}
+          onChange={(e) => searchClubs(e)}
+          className="search-input"
+        />
+        <span className="reset-filters" onClick={() => click()}>
+            reset filters
+        </span>
```
Comment (allenli75) [inline src/components/layout/searchbar/Searchbar.js:17]:
```suggestion
        { showReset && <span className="reset-filters" onClick={() => click()}>
            reset filters
        </span> }
```

File: src/components/layout/searchbar/Searchbar.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from 'react';
+import '../searchbar/Searchbar.scss';
+
+const Searchbar = ({click, value, searchClubs}) => {
```
Comment (allenli75) [inline src/components/layout/searchbar/Searchbar.js:4]:
```suggestion
const Searchbar = ({click, value, searchClubs, showReset}) => {
```

File: src/pages/catalog/Catalog.js [hunk 3]
```diff
@@ -132,20 +133,7 @@ const Catalog = ({
...
-                placeholder="Search"
-                value={name}
-                onChange={(e) => changeSearch(e)}
-                className="search-input"
-              />
-              <span className="reset-filters" onClick={() => resetFilters()}>
-                reset filters
-              </span>
-            </div>
+            <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters}/>
```
Comment (allenli75) [inline src/pages/catalog/Catalog.js:136]:
```suggestion
            <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters} showReset = {true}/>
```

File: src/pages/student/AppTracker.js [hunk 4]
```diff
@@ -98,18 +99,7 @@ const AppTracker = ({
...
-          <input
-            type="text"
-            placeholder="Search"
-            value={name}
-            onChange={(e) => changeSearch(e)}
-            className="search-input"
-          />
-          <span className="reset-filters">reset filters</span>
-        </div>
+        <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters}/>
```
Comment (allenli75) [inline src/pages/student/AppTracker.js:102]:
```suggestion
        <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters} showReset = {false}/>
```

File: src/pages/student/Bookmarks.js [hunk 5]
```diff
@@ -250,18 +251,7 @@ function Bookmarks({
...
-            <input
-              type="text"
-              placeholder="Search"
-              value={name}
-              onChange={(e) => changeSearch(e)}
-              className="search-input"
-            />
-            <span className="reset-filters">reset filters</span>
-          </div>
+          <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters}/>
```
Comment (allenli75) [inline src/pages/student/Bookmarks.js:254]:
```suggestion
          <Searchbar value={name} searchClubs = {changeSearch} click = {resetFilters} showReset = {true}/>
```

**Review by allenli75 — CHANGES_REQUESTED** [review]
hi @jeyu21 suggested some changes! i saw your comment about the search on the modal looking sus and i assumed you were referring to the reset button but lmk if that's wrong.

the reset button shouldn't appear on the application tracker modal anyway so i think it's fine that it looks weird for now. it looks that way because the "reset filters" button is a span with no margins...

tim has a ticket this week to work on building the entire modal and karen is working on refactoring the entire filter bar component so it should be sorted out eventually, but what we can do rn is make the reset button conditionally rendered (the suggestions should take care of that but if you want to learn more about conditional rendering: https://reactjs.org/docs/conditional-rendering.html)

**Comment by jeyu21** [comment]
Thanks for the comments and commits. Am I able to squash and merge now or should I wait the other developers' code?

**Review by allenli75 — APPROVED** [review]
looks great thank you for addressing the conditional rendering suggestions jenny! can go ahead and squash + merge this PR :)

**Pull request merged**